workbench-core/user-management/src/plugins/cognitoUserManagementPlugin.ts

```diff
@@ -525,9 +527,18 @@ export class CognitoUserManagementPlugin implements UserManagementPlugin {
    * @throws {@link UserNotFoundError} if the user provided doesn't exist in the user pool
    * @throws {@link RoleNotFoundError} if the group provided doesn't exist in the user pool
    * @throws {@link TooManyRequestsError} if the RPS limit was exceeded
+   * @throws {@link UserRolesExceedLimitError} if user has reached Cognito group limit
    */
   public async addUserToRole(id: string, role: string): Promise<void> {
     try {
+      // check if user has reached role limits
+      const existingRoles = await this.getUserRoles(id);
```

Comment:
Cognito has a set limit of 100 groups per user. Is it necessary for this additional call? RPS limit on this call is 50

Reply:
Please refer to the ticket issues/P88136850: we are enforcing only 25 groups at this point and throw 4XX instead of 500 error